### PR TITLE
feat(818): Update chart axis values

### LINKED ISSUE #818: Trend charts use an evenly spaced categorical X-axis instead of true time spacing
On several trend charts, the X-axis appears to treat dates as categories rather than points in time.

### Problem
Data points are currently spaced evenly across the chart regardless of the actual interval between measurements.

This means:

two results taken 1 day apart can appear the same distance apart as two results taken 6 months apart
clustered measurements and long gaps are not shown accurately
the visual trend can be misleading, especially for irregularly sampled medical data
Expected behaviour
Trend charts should use a true time axis so spacing reflects the real interval between measurements.

### Likely cause
The chart data appears to pass date strings directly to Recharts, which causes the X-axis to behave like a categorical axis rather than a numeric time scale.

### Suggested fix
Use a timestamp-based X-axis instead of date strings, for example:

convert measurement dates to numeric timestamps
configure XAxis with:
type="number"
scale="time"
Areas affected
This seems relevant at least to:

lab result trend charts
vital trend charts
### Why this matters
For medical data, the timing between observations is clinically relevant. A true time axis gives a more accurate and trustworthy visual representation of the trend.

### PR DESCRIPTION
**Pull request by robf17**
## Summary

Trend charts for lab results and vitals used a categorical X-axis, spacing every data point equally regardless of actual time between measurements. This PR switches all affected `XAxis` instances to a numeric time scale (`type="number" scale="time"`) so the distance between points reflects real elapsed time.

## Related issue

Closes #818

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Refactor / code quality
- [ ] Documentation
- [ ] Translation / i18n
- [ ] Other

## Testing
- 8 new unit tests added in `vitalDataAggregation.test.ts` covering: empty input, timestamp correctness, field mapping, ordering, single-point safety, and secondary value fields
- Edge cases caught during self-review and fixed: single-point axis collapse (±1 day domain padding), aggregated vitals period labels not rendering on axis ticks (forced `ticks` prop), timezone date-shift on `YYYY-MM-DD` lab dates (`T00:00:00` local-midnight fix)

## Checklist

- [x] Tests added or updated (or N/A with reason)
- [x] `npm run lint` and `npm run build` pass
- [x] Backend tests pass (`pytest`) if backend code changed
- [x] No PHI, secrets, or `console.log` left in the diff
- [x] User-facing strings go through `t()` translations
- [x] Documentation updated if API, schema, or service behavior changed

## Summary by CodeRabbit

* **Bug Fixes**
  * Medical vitals and lab-result charts now use numeric time-based x-axes, expanded domains (±1 day) for better date positioning, and improved tick formatting to show localized short dates while preserving aggregated period labels.
  * Qualitative chart behavior adjusted to work with time-scale x-axes.

* **Tests**
  * Added automated tests verifying conversion of aggregated data into chart-ready objects, including secondary value mappings and timestamp handling.

### CODE REVIEW

File: frontend/src/components/medical/vitals/VitalTrendChart.tsx [hunk 1]
```diff
@@ -121,6 +121,7 @@ const VitalTrendChart: React.FC<VitalTrendChartProps> = ({
 
         return {
           date: dateStr,
+          timestamp: new Date(point.recorded_date).getTime(),
```
Comment (afairgiant) [inline frontend/src/components/medical/vitals/VitalTrendChart.tsx:125]: > ```
  > Verify against current code and only apply if the finding holds.
  >
  > In frontend/src/components/medical/vitals/VitalTrendChart.tsx at line 124 and
  > frontend/src/utils/vitalDataAggregation.ts at line 280, the timestamp is
  > computed via `new Date(point.recorded_date).getTime()` (and
  > `new Date(point.date).getTime()` in the util) without the defensive
  > `T00:00:00` local-midnight fix that was applied to the lab chart in
  > TestComponentTrendChart.tsx (lines 52 and 183) to prevent timezone date-shift
  > on "YYYY-MM-DD" inputs.
  >
  > First, confirm whether `VitalDataPoint.recorded_date` (see
  > frontend/src/components/medical/vitals/types.ts) and
  > `AggregatedDataPoint.date` can ever be emitted as a date-only
  > "YYYY-MM-DD" string by the backend — check the Pydantic schemas under
  > app/schemas/ for vitals and any serializers in the vitals endpoint/CRUD
  > layer. If either can produce a date-only string, JS `new Date("YYYY-MM-DD")`
  > parses it as UTC midnight, which renders as the previous day for users west
  > of UTC.
  >
  > If confirmed reachable, update both sites to extract the date portion first
  > (substring before 'T' or the first 10 chars) and reparse as local midnight
  > using the same pattern used in TestComponentTrendChart.tsx:
  >   const dateOnly = raw.split('T')[0];
  >   const ts = dateOnly ? new Date(dateOnly + 'T00:00:00').getTime() : NaN;
  > Preserve fields that already contain a full ISO datetime by only applying
  > the fix when the input has no time component, or apply uniformly if that is
  > simpler — verify aggregated data paths still line up with their periodLabel.
  >
  > If the schema guarantees a full datetime, no change is required; leave a
  > one-line comment at each site noting the assumption so the invariant is
  > visible to future readers.
  > ```

Comment (robf17) [reply frontend/src/components/medical/vitals/VitalTrendChart.tsx:125]: Updated comments.

**Review by afairgiant — APPROVED** [review]
